.github/scripts/e2e-failure-triage/triage.py

```
@@ -0,0 +1,1069 @@
+
+    Flaky tests (tests that failed on one attempt but passed on a retry) are
+    excluded. test-retry records both the failed and passed attempts as
+    separate <testcase> entries with the same name. A test is only considered
+    truly failed if it has no passing entry.
+
+    Returns:
+        - List of actionable component failures (deduplicated by component)
+        - Count of skipped non-component failures
+        - Count of skipped flaky failures
+    """
+    root = ET.fromstring(xml_content)
```

Ruff (0.16.1)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)